HW0/P4/P4.py

```diff
@@ -40,38 +59,115 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    # Read in observed data from P4_measurements.txt
+    s_obs = np.loadtxt('P4_measurements.txt', delimiter=',')
+    
+    # Create matrix to fix dimensional stretching 
+    stretch_fix = np.array([
+        [(1/rx), 0, 0],
+        [0, (1/ry), 0],
+        [0, 0, (1/rz)]
+        ])
+    
+    # Calculate approximate position
+    s_appr = np.dot(stretch_fix, s_obs.transpose()).transpose()
+    
+    # Plot observed trajectory
+    ax.plot(s_appr[:,0], s_appr[:,1], s_appr[:,2],
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    # Initialize progation matrix
+    A = np.mat([
+        [1, 0, 0, dt, 0, 0],
+        [0, 1, 0, 0, dt, 0],
+        [0, 0, 1, 0, 0, dt],
+        [0, 0, 0, 1-c*dt, 0, 0],
+        [0, 0, 0, 0, 1-c*dt, 0],
+        [0, 0, 0, 0, 0, 1-c*dt]
+        ])
```

Comment:
Instead of hard coding, you could also construct two arrays and use the np.diag() function to insert the arrays into a zero matrix. This should help if we are recording more parameters and have a huge A matrix. (This comment also applies to later matrix definitions, but I think it's a trivial point)